### PR TITLE
Feat: Clicking an author's name on each page will show a lit of all pages that author creates.

### LINKED ISSUE #110: Clicking an author's name on each page will show a lit of all pages that author creates.
Here is the workflow:
- You are on any page.
- Each page has an author's name of that page.
- An author is a link.
- Clicking an author's name will be redirect to a lit of all pages that author creates.

This feature is useful for an author that later he/she can use it for his/her portfolio.

### PR DESCRIPTION
**Pull request by mildronize**
Please review my PR ;)

Note: Waiting for decorating page

image

### CODE REVIEW

**Comment by mildronize** [comment]
Done for UI

screencapture-localhost-8000-contributors-2021-08-11-00_28_06

File: src/components/MdxComponents/Contributors.tsx [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
+  width: 100%;
+`;
 
 const style = theme => css`
-  display: flex;
-  flex-direction: row;
-  flex-wrap: wrap;
+  .wrapper {
+    display: flex;
+    flex-direction: row;
```
Comment (aaronamm) [inline src/components/MdxComponents/Contributors.tsx:24]: @mildronize just FYI flex-direction is row by default, later you don't need to specify this value.

File: src/components/MdxComponents/Contributors.tsx [hunk 2]
```diff
@@ -31,12 +51,26 @@ const style = theme => css`
...
+    cursor: pointer;
+    padding: 20px;
+
+    &:hover{
+      background-color: #d8d8d8;
+    }
+  }
 `;
 
+const query = typeof window !== 'undefined' ? window.location.search.replace(/^\?/, '') : '';
```
Comment (aaronamm) [inline src/components/MdxComponents/Contributors.tsx:67]: 👍

File: src/components/MdxComponents/Contributors.tsx [hunk 3]
```diff
@@ -52,22 +86,56 @@ const Contributors = () => {
...
-            />
-            <span>{contributor.login}</span>
-          </a>
-        );
-      })
-      }
-    </div>
+    <>
+
+      <div css={style(theme)}>
```
Comment (aaronamm) [inline src/components/MdxComponents/Contributors.tsx:101]: If the theme div is the outmost, you can remove `<> </>`.

File: src/components/MdxComponents/ContributorsDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+    color: ${theme.colors.primary}
+  }
+
+  a:hover {
+    color: ${theme.colors.primaryDark}
+  }
+
+`;
+
+const repo = 'dotnetthailand/dotnetthailand.github.io';
```
Comment (aaronamm) [inline src/components/MdxComponents/ContributorsDetail.tsx:56]: Very minor, please read this value from config.yml which is 
```yml
features:
  editOnRepo:
        repo: dotnetthailand/dotnetthailand.github.io
```

File: src/components/MdxComponents/ContributorsDetail.tsx [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+  logUrl: string;
+}
+
+interface IGithubUser {
+  profileUrl?: string,
+  avatarUrl?: string,
+  name?: string,
+}
+
+const setTimeoutPromise = (timeout: number) => new Promise(resolve => {
```
Comment (aaronamm) [inline src/components/MdxComponents/ContributorsDetail.tsx]: Please use our existing `sleep` function.

File: src/components/MdxComponents/ContributorsDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+    await setTimeoutPromise(delayTime * 1000);
+    return fetchRetry(url, delayTime, limit - 1);
+  }
+}
+
+const isContent = (filename: string) => {
+  return /^content/.test(filename);
+}
+
+const isIgnoreFile = (filename: string) => {
```
Comment (aaronamm) [inline src/components/MdxComponents/ContributorsDetail.tsx:114]: It think this function can be written with some function.

**Review by aaronamm — CHANGES_REQUESTED** [review]
I have only some minor feedbacks.

You did a good job.

Thank you so much.

File: src/components/MdxComponents/Contributors.tsx (hunk 1 above)
Comment (mildronize) [reply src/components/MdxComponents/Contributors.tsx:24]: The existing code use `flex-direction: row;` , No need to edit, I just refactor it.

File: src/components/MdxComponents/Contributors.tsx (hunk 2 above)
Comment (mildronize) [reply src/components/MdxComponents/Contributors.tsx:67]: Thank you

File: src/components/MdxComponents/Contributors.tsx (hunk 3 above)
Comment (mildronize) [reply src/components/MdxComponents/Contributors.tsx:101]: Thank you

File: src/components/MdxComponents/ContributorsDetail.tsx (hunk 4 above)
Comment (mildronize) [reply src/components/MdxComponents/ContributorsDetail.tsx:56]: The state management is terrible,  drill down the prop is required. No fix.

File: src/components/MdxComponents/ContributorsDetail.tsx (hunk 5 above)
Comment (mildronize) [reply src/components/MdxComponents/ContributorsDetail.tsx]: OK

File: src/components/MdxComponents/ContributorsDetail.tsx (hunk 6 above)
Comment (mildronize) [reply src/components/MdxComponents/ContributorsDetail.tsx:114]: Yes, can be improved later.

**Comment by mildronize** [comment]
@aaronamm Thank you so much .

**Review by aaronamm — APPROVED** [review]

**Pull request merged**